articles/active-directory/develop/quickstarts/active-directory-windesktop.md

```diff
@@ -0,0 +1,142 @@
+>
+> ### Option 1: Register and auto configure your app and then download your code sample
+>
+> 1. Go to the [Azure portal - Application Registration](https://portal.azure.com/signin/index/?Microsoft_AAD_RegisteredApps=true#blade/Microsoft_AAD_RegisteredApps/applicationsListBlade/quickStartType/WinDesktopQuickstartPage/sourceType/docs).
+> 1. Enter a name for your application and click **Register**.
+> 1. Follow the instructions to download and automatically configure your new application for you in one click.
+>
+> ### Option 2: Register and manually configure your application and code sample
+>
+> #### Step 1: Register your application
+> 
+> 1. To register an application for this code sample, go to the [Azure portal - Application Registration](https://aka.ms/registeredappsprod) and select **New registration**.
+> 1. Enter a name for your application, add `urn:ietf:wg:oauth:2.0:oob` in **Reply URL**, and click **Register**.
+
+> [!div renderon="portal" class="sxs-lookup"]
+> #### Step 1: Configure your application
+> For the code sample for this quickstart to work, you need to add a reply URL as **urn:ietf:wg:oauth:2.0:oob**.
+> > [!div renderon="portal" id="makechanges" class="nextstepaction"]
+> > [Make this change for me]()
+>
+> > [!div id="appconfigured" class="hidden"]
+> > ![Already configured](../../../../includes/media/active-directory-develop-common/green-check.png) Your application is configured with these attributes
+
+#### Step 2: Download your Visual Studio project
+
+ [Download the Visual Studio 2017 project](https://github.com/Azure-Samples/active-directory-dotnet-desktop-msgraph-v2/archive/master.zip)
+
+#### Step 3: Configure your Visual Studio project
+
+ 1. Open the project in Visual Studio
+ 1. Edit **App.Xaml.cs** and replace the line starting with `private static string ClientId` with the Application ID from the application you just registered:
+
+    ```csharp
+    private static string ClientId = "Enter_the_Application_Id_here";
+    ```
+
+## More Information
+
+### MSAL.NET
+
+MSAL ([Microsoft.Identity.Client](https://www.nuget.org/packages/Microsoft.Identity.Client)) is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. You can install it by running the following command in Visual Studio's *Package Manager Console*:
+
+```powershell
+Install-Package Microsoft.Identity.Client -Pre
+```
+
+### MSAL Initialization
+
+You can add the reference for MSAL by adding the line below:
+
+```csharp
+using Microsoft.Identity.Client;
+```
+
+Then initialize MSAL using the line below:
+
+```csharp
+public static PublicClientApplication PublicClientApp = new PublicClientApplication(ClientId);
+```
+
+> |Where: ||
+> |---------|---------|
+> |ClientId | The Application Id from the application registered in *portal.microsoft.com* |
+
+### Requesting tokens
+
+Msal has two methods used acquire tokens - `AcquireTokenAsync` and `AcquireTokenSilentAsync`:
```

Comment:
Again, this is `PublicClientApplication`, which has two methods for the moment. By the time of Ignite, there will be more methods, BTW (WIA, username/password). "**PublicClientApplication** has two methods used to acquire **tokens interactively**"